File: server/api/routes/payment_routes.py
from fastapi import APIRouter, Body, HTTPException
from controllers.paymentController import create_checkout_session, create_payment_intent

router = APIRouter()


@router.post("/create-checkout-session")
async def create_checkout_session_route(body: dict = Body(...)):
    try:
        return await create_checkout_session(body)
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/create-payment-intent")
async def create_payment_intent_route(body: dict = Body(...)):
    try:
        return await create_payment_intent(body)
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: server/api/routes/paymentRoutes.js
const express = require('express');
const { createCheckoutSession, webhookHandler, getStripeStatus, createPaymentIntent } = require('../controllers/paymentController.js');

const router = express.Router();

// Public endpoints - allow both guest and authenticated users
router.post('/create-checkout-session', createCheckoutSession);

// Status (debug) - shows whether Stripe key is configured (masked preview)
router.get('/status', getStripeStatus);

// Create PaymentIntent for Elements (in-page) flow - allow both guest and authenticated
router.post('/create-payment-intent', createPaymentIntent);

// Note: webhook should be mounted with raw body - Server.js mounts the raw endpoint.

module.exports = router;

